### PR TITLE
Module7 2 full project

### PR DESCRIPTION
**Pull request by anna-kurmaz**

### CODE REVIEW

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/EmployeeHibernateDAO.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public Employee getById(long id) {
+        return sessionFactory.getCurrentSession().get(Employee.class, id);
+    }
+
+    @Override
+    public List<Employee> getByName(String name) {
+        Session session = sessionFactory.getCurrentSession();
+        Query query = session.createQuery("Select v from Employee v where name like :name");
```
Comment (schaffe) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/EmployeeHibernateDAO.java]: Why Employee is "v"?

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.List;
+
+public class IngredientHibernateDAO implements DAO<Ingredient> {
+
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void save(Ingredient value) {
+
```
Comment (schaffe) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java]: empty methods? If you don`t need them better `throw UnsupportedOperationException" instead. Or provide such default implementation in interface.

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public List<Ingredient> getByName(String name) {
+        return null;
+    }
+
+    @Override
+    public List<Ingredient> getAll() {
+        Session session = sessionFactory.getCurrentSession();
```
Comment (schaffe) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java]: Where do you close session?

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/model/Dish.java [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+import javax.persistence.*;
 import java.util.ArrayList;
 import java.util.List;
 
+@Entity
+@Table(name = "DISH")
 public class Dish {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (schaffe) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/model/Dish.java]: It`s better to use UUID for ids

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/model/Ingredient.java [hunk 5]
```diff
@@ -1,9 +1,20 @@
 package com.homework.model;
 
+import javax.persistence.*;
+
+@Entity
+@Table(name = "INGREDIENT")
 public class Ingredient {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private long id;
```
Comment (schaffe) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/model/Ingredient.java]: use boxed types is JPA

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/EmployeeHibernateDAO.java (hunk 1 above)
Comment (anna-kurmaz) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/EmployeeHibernateDAO.java]: fixed

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java (hunk 2 above)
Comment (anna-kurmaz) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java]: fixed

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java (hunk 3 above)
Comment (anna-kurmaz) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/dao/hibernate/IngredientHibernateDAO.java]: found in description that session opened with "getCurrentSession()" will be closed atomatically (unlike openSession())

File: Enterprise/Module6_2_Restaurant/src/main/java/com/homework/model/Ingredient.java (hunk 5 above)
Comment (anna-kurmaz) [inline Enterprise/Module6_2_Restaurant/src/main/java/com/homework/model/Ingredient.java]: fixed

**Comment by schaffe** [comment]
Able to merge

**Pull request closed without merging**